In [24]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load in 

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the "../input/" directory.
# For example, running this (by clicking run or pressing Shift+Enter) will list the files in the input directory

import os
print(os.listdir("../input"))

# Any results you write to the current directory are saved as output.

['train.csv', 'sample_submission.csv', 'test.csv', 'data_description.txt']


# Importing the data:

In [25]:
df_train = pd.read_csv('../input/train.csv')
df_train.head()

Id  MSSubClass MSZoning    ...     SaleType  SaleCondition SalePrice
0   1          60       RL    ...           WD         Normal    208500
1   2          20       RL    ...           WD         Normal    181500
2   3          60       RL    ...           WD         Normal    223500
3   4          70       RL    ...           WD        Abnorml    140000
4   5          60       RL    ...           WD         Normal    250000

[5 rows x 81 columns]

> First thing noticeable from the top 5 rows is, features 'Alley', 'PoolQC', 'Fence' & 'MiscFeature' have all 5 of the values missing.
# Describing the training dataset:

In [26]:
df_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
Id               1460 non-null int64
MSSubClass       1460 non-null int64
MSZoning         1460 non-null object
LotFrontage      1201 non-null float64
LotArea          1460 non-null int64
Street           1460 non-null object
Alley            91 non-null object
LotShape         1460 non-null object
LandContour      1460 non-null object
Utilities        1460 non-null object
LotConfig        1460 non-null object
LandSlope        1460 non-null object
Neighborhood     1460 non-null object
Condition1       1460 non-null object
Condition2       1460 non-null object
BldgType         1460 non-null object
HouseStyle       1460 non-null object
OverallQual      1460 non-null int64
OverallCond      1460 non-null int64
YearBuilt        1460 non-null int64
YearRemodAdd     1460 non-null int64
RoofStyle        1460 non-null object
RoofMatl         1460 non-null object
Exterior1st      1460 non-n

In [27]:
df_train.describe()

Id   MSSubClass      ...             YrSold      SalePrice
count  1460.000000  1460.000000      ...        1460.000000    1460.000000
mean    730.500000    56.897260      ...        2007.815753  180921.195890
std     421.610009    42.300571      ...           1.328095   79442.502883
min       1.000000    20.000000      ...        2006.000000   34900.000000
25%     365.750000    20.000000      ...        2007.000000  129975.000000
50%     730.500000    50.000000      ...        2008.000000  163000.000000
75%    1095.250000    70.000000      ...        2009.000000  214000.000000
max    1460.000000   190.000000      ...        2010.000000  755000.000000

[8 rows x 38 columns]

>There are a total of 81 features and 1460 samples.
# Handling the missing data

In [28]:
df_train.isnull().sum().sort_values(ascending=False)

PoolQC           1453
MiscFeature      1406
Alley            1369
Fence            1179
FireplaceQu       690
LotFrontage       259
GarageCond         81
GarageType         81
GarageYrBlt        81
GarageFinish       81
GarageQual         81
BsmtExposure       38
BsmtFinType2       38
BsmtFinType1       37
BsmtCond           37
BsmtQual           37
MasVnrArea          8
MasVnrType          8
Electrical          1
Utilities           0
YearRemodAdd        0
MSSubClass          0
Foundation          0
ExterCond           0
ExterQual           0
Exterior2nd         0
Exterior1st         0
RoofMatl            0
RoofStyle           0
YearBuilt           0
                 ... 
GarageArea          0
PavedDrive          0
WoodDeckSF          0
OpenPorchSF         0
3SsnPorch           0
BsmtUnfSF           0
ScreenPorch         0
PoolArea            0
MiscVal             0
MoSold              0
YrSold              0
SaleType            0
Functional          0
TotRmsAbvGrd        0
KitchenQua

Droping the features with missing values more than 81. Since they are missing a lot of data.

In [29]:
df_train = df_train.drop(['PoolQC', 'MiscFeature', 'Alley', 'Fence', 'FireplaceQu', 'LotFrontage'], axis=1)

* The minimum value of (TotalBsmtSF: Total square feet of basement area) is 0, which means some of the houses does'nt have a basement and probably that's why we have missing values in 'BsmtExposure', 'BsmtFinType2' 'BsmtFinType1', 'BsmtCond', 'BsmtQual' features.
* Similarly, the minimum value of (GarageArea: Size of garage in square feet) is 0, which means some of the houses does'nt have a garage and probably that's why we have missing values in 'GarageCond', 'GarageType', 'GarageYrBlt', 'GarageFinish', 'GarageQual' features.
* Missing Values will be interpolated by the median of that feature, as it is suitable for continuous data with outliers

In [ ]:
# var = ['GarageCond', 'GarageType', 'GarageYrBlt', 'GarageFinish', 'GarageQual', 'BsmtExposure',
#        'BsmtFinType2' 'BsmtFinType1', 'BsmtCond', 'BsmtQual', 'MasVnrArea', 'MasVnrType', 'Electrical']
# data = pd.concat([df_train['SalePrice'], df_train[var[0]]], axis=1)
# data.plot.scatter(x=var[0], y='SalePrice', ylim=(0,800000));